ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@rossberg-chromium If we decoupled construction of tuples of values from `br` (such that you'd write `(br_if $1 (tuple a b) (cond))` or just `(br_if $1 (call foo_returning_tuple) (cond))`, then is there a problem?  In the model of this I have in my head, a tuple would just be 1 entry on the expr stack (so that means each entry on the expr stack is a `Vector<Pair<Def,Type>>` where `Def` is your compiler SSA definition) so you still want the binary `br0`/`br` split.